test/integration/timeout/timeout_stack.go

```diff
@@ -0,0 +1,666 @@
+package timeout
+
+import (
+	"context"
+	"fmt"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/manifest"
+	"io/ioutil"
+	"net/http"
+
+	appmesh "github.com/aws/aws-app-mesh-controller-for-k8s/apis/appmesh/v1beta2"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/pkg/algorithm"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework"
+	"github.com/aws/aws-app-mesh-controller-for-k8s/test/framework/k8s"
+	"github.com/aws/aws-sdk-go/aws"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	defaultFrontEndImage = "928111597794.dkr.ecr.us-west-2.amazonaws.com/amazon/timeout-e2e:feapp-final"
+	defaultBackEndImage  = "928111597794.dkr.ecr.us-west-2.amazonaws.com/amazon/timeout-e2e:colorapp-final-1"
+
+	timeoutTest      = "timeout-e2e"
+	AppContainerPort = 8080
+
+	timeoutMessage          = "upstream request timeout"
+	expectedBackendResponse = "backend"
+)
+
+// Timeout stack is setup as below
+//	FrontEnd ->
+//        - Exposes two endpoints "/default" and "/timeout" and reaches out to backend for both the endpoints
+//        - Refers to "backend" VirtualService which in turn refers to "backend" VirtualRouter
+//
+//  Backend ->
+//        - Exposes "/default" and "/timeout" with a configured 45 seconds delay
+//        - "backend" VN is configured with a Listener timeout of 60 seconds
+//        - "backend" VR has two routes "/default" and "/timeout"
+//        - "/default" path uses default timeout of 15s and "/timeout" path is configured with 60s timeout
```

Comment:
The description of the stack is very helpful. Thanks for adding it!

The test relies of hosted images and part of the logic relies on the application. Can we please:
1) commit the application code in this suite so it's all contained here. Don't have to add build->publish->run in this suite support yet - we can add that later
2) remove the fixed time out from the application code and make the time out values picked up as part of env variables in the application that you can define as part of Pod spec